internal/utils/utils.go

```diff
@@ -40,36 +46,40 @@ func GetArchitecture() string {
 // DownloadFile will download a url and store it in a temporary path.
 // It writes to the destination file as it downloads it, without
 // loading the entire file into memory.
-func DownloadFile(url string) (string, error) {
-	tempParentDir := filepath.Join(os.TempDir(), uuid.NewString())
-	internalio.MkdirAll(tempParentDir)
+func DownloadFile(downloadRequest *DownloadRequest) error {
+	var filePath string
+	if downloadRequest.DownloadPath == "" {
+		tempParentDir := filepath.Join(os.TempDir(), uuid.NewString())
+		internalio.MkdirAll(tempParentDir)
+		filePath = filepath.Join(tempParentDir, uuid.NewString())
```

Comment:
Should we update the downloadPath to the temp file? I wonder if consumers would like to use it after downloading
```suggestion
		filePath = filepath.Join(tempParentDir, uuid.NewString())
		downloadRequest.DownloadPath = filePath
```